vgridjs/src/timeline_track.tsx

```diff
@@ -106,6 +136,69 @@ class TimelineBounds {
   }
 }
 
+interface TimelineNavigatorProps {
+  timeline_bounds: TimelineBounds,
+  timeline_width: number,
+  full_duration: number
+}
+
+@canvas_component
+class TimelineNavigator extends React.Component<TimelineNavigatorProps, {}> {
+  private canvas_ref : React.RefObject<HTMLCanvasElement>;
+  private disposer : any;
+  
+  constructor (props: TimelineNavigatorProps) {
+    super(props);
+    this.canvas_ref = React.createRef();
+  }
+  
+  handleChange = (event: any) => {
+    if (event.target) {
+      let value  = event.target.valueAsNumber;
+      let max = parseInt(event.target.max);
+      let start = (this.props.full_duration * (value / this.props.timeline_width)) - (this.props.timeline_bounds.span()/2);
+      let end = (this.props.full_duration * (value / this.props.timeline_width)) + (this.props.timeline_bounds.span()/2);
+
+      //Make sure that start/end of bounds do not extend past the duration of the video
+      if (start < 0) {
+        end -= start;
+        start = 0;
+      }
+      else if (end > this.props.full_duration) {
+        start -= end - this.props.full_duration;
+        end = this.props.full_duration
+      }
+      this.props.timeline_bounds.set_bounds(start, end);
+      this.forceUpdate();
```

Comment:
`forceUpdate` seems suspicious. Is the issue that the `TimelineNavigator` is not marked as `@observer?`